File: Inversion/serial/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/flib/gclip.f
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: CLIP                                                  c
c                                                                     c
c      COPYRIGHT (C)  1997 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      3507 Laclede Avenue                                            c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
        subroutine  gclip(cmd,  xlow, ylow, xhigh, yhigh )
        character cmd*(*)
        real*4 xlow, ylow, xhigh, yhigh
c-----
c       Set clip region
c       cmd Char*(*)    "ON" or "OFF"
c       xlow    R*4     Coordinates of bounding box
c       ylow    R*4     
c       xhigh   R*4     
c       yhigh   R*4     
c-----
        common/Xcplot/xold,yold,xcur,ycur
        common/Ycplot/xstp,ystp

        integer*4 ilw, jlw, iup, jup
        integer*4 icmd
        xx = 1000. * (xlow*xstp + xold)
        yy = 1000. * (ylow*xstp + yold)
        if(xx .gt. 1000000000.0)then
            ilw = 1000000000
        else if(xx .lt. -1000000000.0)then
            ilw = -1000000000
        else
            ilw = xx 
        endif
        if(yy .gt. 1000000000.0)then
            jlw = 1000000000
        else if(yy .lt. -1000000000.0)then
            jlw = -1000000000
        else
            jlw = yy 
        endif
        xx = 1000. * (xhigh*xstp + xold)
        yy = 1000. * (yhigh*xstp + yold)
        if(xx .gt. 1000000000.0)then
            iup = 1000000000
        else if(xx .lt. -1000000000.0)then
            iup = -1000000000
        else
            iup = xx 
        endif
        if(yy .gt. 1000000000.0)then
            jup = 1000000000
        else if(yy .lt. -1000000000.0)then
            jup = -1000000000
        else
            jup = yy 
        endif
        if(cmd(1:1) .eq. 'o' .or. cmd(1:1) .eq. 'O')then
            if(cmd(2:2) .eq. 'n' .or. cmd(2:2) .eq. 'N')then
                icmd = 1
            else
                icmd = 0
            endif
        else
            icmd = 0
        endif
        call dfclip(icmd, ilw, jlw, iup, jup)
        return
        end

File: Inversion/serial/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/clib/gclip.c
#include "callib.h"
#include "dosubs.h"
#include "calplot.h"
extern struct Gcplot Gcplt;

void gclip(char *cmd, float xlow,float ylow,float xhigh,float yhigh )
{
	float xx, yy;
	INT ilw, jlw, iup, jup;
	INT icmd;
       	xx = 1000. * (xlow*Gcplt.xstp + Gcplt.xold);
       	yy = 1000. * (ylow*Gcplt.ystp + Gcplt.yold);
	if(xx > 1000000000.0)
		ilw = 1000000000;
	else if(xx < -1000000000.0)
		ilw = -1000000000;
	else
		ilw = xx ;
	if(yy > 1000000000.0)
		jlw = 1000000000;
	else if(yy < -1000000000.0)
		jlw = -1000000000;
	else
		jlw = yy ;
       	xx = 1000. * (xhigh*Gcplt.xstp + Gcplt.xold);
       	yy = 1000. * (yhigh*Gcplt.ystp + Gcplt.yold);
	if(xx > 1000000000.0)
		iup = 1000000000;
	else if(xx < -1000000000.0)
		iup = -1000000000;
	else
		iup = xx ;
	if(yy > 1000000000.0)
		jup = 1000000000;
	else if(yy < -1000000000.0)
		jup = -1000000000;
	else
		jup = yy ;
	if(cmd[0] == 'o' || cmd[0] == 'O'){
		if(cmd[1] == 'n' || cmd[1] == 'N')
			icmd = 1;
		else
			icmd = 0;
	} else
		icmd = 0;
	(*do_clip)(icmd, ilw, jlw, iup, jup);
}
